toolbox/ui.py

```diff
@@ -136,7 +136,28 @@ def draw_umap_projections(self, utterances: Set[Utterance]):
         self.umap_ax.set_xticks([])
         self.umap_ax.set_yticks([])
         self.umap_ax.figure.canvas.draw()
-        
+
+    def setup_audio_devices(self,sample_rate):
+        supported_devices=[]
+        for device in sd.query_devices():
+            try:
+                sd.check_output_settings(device=device['name'], samplerate=sample_rate)
+                supported_devices.append(device['name'])
+            except Exception as e:
+                warn("Unsupported device "+ device['name']+" for the sample rate: "+str(sample_rate)+": "+str(e))
+        if len(supported_devices)==0:
+            error_dialog = QtWidgets.QErrorMessage()
+            error_dialog.showMessage("No supported output audio devices were found! Audio output may not work.")
+            self.audio_out_devices_cb.setDisabled(True)
+        else:
+            sd.default.device = supported_devices[0]
```

Comment:
sd.default.device sets both input and output devices. On my system, this is overriding my input device and making me unable to record.

A better way to do it here would be:
```
sd.default.device = (None, supported_devices[0]) #(input, output)
```

If default.device is set to None it will default to whatever portaudio returns.